### ISSUE TITLE
Real world test issue

### ISSUE THREAD

**Issue by syt06007** [YingqianWang/OACC-Net#7]
Hello,

I really appreciate your open source code.

I have encountered a few issues while using the code. I have some questions regarding the real-world test. Are there specific hyperparameters or settings that you used? I am having trouble with the reconstruction using the Stanford dataset (Lego Knights and Bulldozer) as described in your paper.

I downloaded the data from this link: http://lightfield.stanford.edu/lfs.html. After obtaining the "Rectified and cropped images," I used a 17x17 image and input only the central 9x9 view into the model(crop, patchsize=128).

Thank you.
![output](https://github.com/user-attachments/assets/0ee5b3b1-3750-4693-9ec3-799f7ae7d1b0)

**Comment by QianChengLiu6** [comment]
Hello, senior
 I also encountered a similar problem. May I ask if your issue has been resolved？

**Comment by syt06007** [comment]
I had to resize test images to the same size as the training data (e.g., if training size was 512x512, test images must also be 512x512).

I'm not sure why this impacts the testing accuracy, especially test.py script already sets a specific patch size (i.e., it slices the input) for the model.

**Comment by QianChengLiu6** [comment]
Hello, senior

I hope this message finds you well.
Thank you very much for your previous kind reply — your guidance was truly helpful to me.

I recently tested the author’s open-source code using publicly available images (resized to 512×512). However, the results I obtained were still not satisfactory. Compared with the images shown in the original paper, my output is obviously much worse.

![Image](https://github.com/user-attachments/assets/034c9b9e-0cd6-4596-ab5b-fd204946bf22)

![Image](https://github.com/user-attachments/assets/25f1f0ce-119f-44e2-9728-e5f162e8ec2a)

May I ask if your reproduced results could reach the same quality as those presented in the paper? I wonder if I might have missed some important steps during the process. If it is convenient for you, I would be very grateful for any advice or guidance you could kindly provide.

Thank you again for your time and consideration.
I sincerely appreciate your help and wish you continued success in your research.

Warm regards

**Comment by syt06007** [comment]
```
import argparse
from utils import *
from model import Net
import cv2

def parse_args():
    parser = argparse.ArgumentParser()
    parser.add_argument('--device', type=str, default='cuda:0')
    parser.add_argument("--angRes", type=int, default=9, help="angular resolution")
    parser.add_argument('--model_name', type=str, default='OACC-Net')
    # parser.add_argument('--testset_dir', type=str, default='./Data/LF_real/')
    parser.add_argument('--testset_dir', type=str, default='./OACC-Net/dataset/validation/')
    parser.add_argument('--crop', type=bool, default=True)
    parser.add_argument('--patchsize', type=int, default=128)
    parser.add_argument('--minibatch_test', type=int, default=4)
    parser.add_argument('--model_path', type=str, default='./log/OACC-Net.pth.tar')
    parser.add_argument('--save_path', type=str, default='./')
    return parser.parse_args()

'''
Note: 1) We crop LFs into overlapping patches to save the CUDA memory during inference. 
      2) When cropping is performed, the inference time will be longer than the one reported in our paper.
'''

def test(cfg):

    scene_list = os.listdir(cfg.testset_dir)
    angRes = cfg.angRes

    net = Net(cfg.angRes)
    net.to(cfg.device)
    model = torch.load(cfg.model_path, map_location={'cuda:0': cfg.device})
    net.load_state_dict(model['state_dict'])

    for scenes in scene_list:
        print('Working on scene: ' + scenes + '...')
        temp = imageio.imread(cfg.testset_dir + scenes + '/input_Cam000.png')
        
        new1=512
        aspect_ratio = new1/temp.shape[1]
        new0 = int(temp.shape[0]*aspect_ratio)
        temp = cv2.resize(temp, (new0, new1))
        
        # temp = cv2.resize(temp, (512, 512))
        lf = np.zeros(shape=(9, 9, new0, new1, 3), dtype=int)
        # lf = np.zeros(shape=(9, 9, temp.shape[0], temp.shape[1], 3), dtype=int)
        for i in range(81):
            temp = imageio.imread(cfg.testset_dir + scenes + '/input_Cam0%.2d.png' % i)
            temp = cv2.resize(temp, (new1, new0))


            # temp = cv2.resize(temp, (512, 512))
            lf[i // 9, i - 9 * (i // 9), :, :, :] = temp
        lf_gray = np.mean((1 / 255) * lf.astype('float32'), axis=-1, keepdims=False)
        angBegin = (9 - angRes) // 2
        lf_angCrop = lf_gray[angBegin:  angBegin + angRes, angBegin: angBegin + angRes, :, :]

        if cfg.crop == False:
            data = rearrange(lf_angCrop, 'u v h w -> (u h) (v w)')
            data = ToTensor()(data.copy())
            with torch.no_grad():
                disp = net(data.unsqueeze(0).to(cfg.device))
            disp = np.float32(disp[0,0,:,:].data.cpu())

        else:
            patchsize = cfg.patchsize
            stride = patchsize // 2
            data = torch.from_numpy(lf_angCrop)
            sub_lfs = LFdivide(data.unsqueeze(2), patchsize, stride)
            n1, n2, u, v, c, h, w = sub_lfs.shape
            sub_lfs = rearrange(sub_lfs, 'n1 n2 u v c h w -> (n1 n2) u v c h w')
            mini_batch = cfg.minibatch_test
            num_inference = (n1 * n2) // mini_batch
            with torch.no_grad():
                out_disp = []
                for idx_inference in range(num_inference):
                    current_lfs = sub_lfs[idx_inference * mini_batch: (idx_inference + 1) * mini_batch, :, :, :, :, :]
                    input_data = rearrange(current_lfs, 'b u v c h w -> b c (u h) (v w)')
                    out_disp.append(net(input_data.to(cfg.device)))

                if (n1 * n2) % mini_batch:
                    current_lfs = sub_lfs[(idx_inference + 1) * mini_batch:, :, :, :, :, :]
                    input_data = rearrange(current_lfs, 'b u v c h w -> b c (u h) (v w)')
                    out_disp.append(net(input_data.to(cfg.device)))

            out_disps = torch.cat(out_disp, dim=0)
            out_disps = rearrange(out_disps, '(n1 n2) c h w -> n1 n2 c h w', n1=n1, n2=n2)
            disp = LFintegrate(out_disps, patchsize, patchsize // 2)
            disp = disp[0: data.shape[2], 0: data.shape[3]]
            disp = np.float32(disp.data.cpu())

        print('Finished! \n')
        write_pfm(disp, cfg.save_path + '%s.pfm' % (scenes))

    return


if __name__ == '__main__':
    cfg = parse_args()
    test(cfg)

```
I’m not entirely sure, but I believe I also used this code (test.py).
Unfortunately, the PFM files for images other than Lego Knights are no longer available, so please try running this code yourself and let me know the results.

<img width="895" height="582" alt="Image" src="https://github.com/user-attachments/assets/0fb58366-802a-46b1-8a56-3142bcfc26c8" />

**Comment by QianChengLiu6** [comment]
Hello, senior

Thank you again for your patient reply.

I have sent the results and related codes to your personal email([32181265@*******.ac.kr](mailto:32181265@dankook.ac.kr)). Please check it



Thank you very much.

**Comment by GaoR311** [comment]
> Hello, senior
> 
> Thank you again for your patient reply.
> 
> I have sent the results and related codes to your personal email([32181265@*******.ac.kr](mailto:32181265@dankook.ac.kr)). Please check it
> 
> Thank you very much.

Do you have solved this problem?I also can't get a good result in stanford .My result is similar with you.If you have solved ,please tell me,thank you very much
